pkg/image/registryclient/client.go

```diff
@@ -226,6 +240,32 @@ func (c *Context) Repository(ctx context.Context, registry *url.URL, repoName st
 	return NewLimitedRetryRepository(repo, c.Retries, limiter), nil
 }
 
+// PreferredImageSource takes a list of alternative image sources and returns an appropriate image reference, repository, and manifest service  for a given Context, or an error
+func (c *Context) PreferredImageSource(ctx context.Context, image imagereference.DockerImageReference, alternativeSources []imagereference.DockerImageReference, insecure bool) (imagereference.DockerImageReference, distribution.Repository, distribution.ManifestService, error) {
```

Comment:
Note: No reference to operatorv1..ICSP

Reply:
If no alternative sources, try the given image ref and return (no change)

Reply:
Yeah, but you're still exposing completely new method, instead of embedding `c.ImageSource != nil` and reading alternatives in `Repository` method.

Reply:
_hm...._